src/openassetio-python/CMakeLists.txt

```diff
@@ -67,16 +67,28 @@ endif ()
 
 if (OPENASSETIO_ENABLE_TESTS AND Python_Interpreter_FOUND)
 
-    # Command to create a Python environment in the install directory,
-    # then install test-specific dependencies (e.g. pytest).
+    # Command to create a Python environment in the install directory.
     add_custom_command(
         OUTPUT
         "${CMAKE_INSTALL_PREFIX}/bin/pip"
-        "${CMAKE_INSTALL_PREFIX}/bin/pytest"
         COMMAND
         "${Python_EXECUTABLE}" -m venv "${CMAKE_INSTALL_PREFIX}"
         COMMAND
+        ${CMAKE_INSTALL_PREFIX}/bin/python -m pip install --upgrade pip
+    )
+    # Target to wrap the venv command so it can be executed in isolation.
+    add_custom_target(
+        openassetio-python-venv
+        DEPENDS "${CMAKE_INSTALL_PREFIX}/bin/pip"
+    )
+
+    # Command to install test-specific dependencies (e.g. pytest).
+    add_custom_command(
+        OUTPUT
+        "${CMAKE_INSTALL_PREFIX}/bin/pytest"
```

Comment:
[question] `CMake` newbie here. Are these like file targets in a `Makefile`, in that, with our flaky python install in the docker image, will this run every time as `bin/pytest` (and `bin/pip`) won't ever end up existing. Not sure there is much we can do about that though?